server/src/server.ts

```diff
@@ -105,30 +107,58 @@ connection.onInitialize(
 );
 
 connection.onInitialized((_params) => {
-    connection.workspace.getConfiguration('yseopml.parseAllProjectFilesAtStartup').then((_confValue) => {
-        parseAllProjectFilesAtStartup = _confValue;
-        if (!parseAllProjectFilesAtStartup) {
-            connection.console.log('Not parsing project files at startup.');
-            return;
-        }
-        connection.console.log('Parsing project files at startup.');
-        connection.workspace.getWorkspaceFolders().then((_value) => {
-            const workspacePath = url.fileURLToPath(_value[0].uri);
-            glob('**/project.kao', {
-                cwd: workspacePath,
-            })
-                .then((_matches) => {
-                    for (const filePath of _matches) {
-                        if (openProjectFile(workspacePath, path.join(workspacePath, filePath))) {
-                            // There should be only one `project.kao` file. If we found a good candidate, stop the loop.
-                            return;
+    connection.workspace
+        .getConfiguration('yseopml.kaoFiles')
+        .then((_kaoFiles) => {
+            // initialize array kaoFilesToParse
+            kaoFilesToParse = _kaoFiles;
+        })
+        .then(() => {
+            connection.workspace.getConfiguration('yseopml.parseAllProjectFilesAtStartup').then((_confValue) => {
```

Comment:
Huuuum isn’t `parseAllProjectFilesAtStartup` a much older option? Why does a huge chunk of code pertaining to it suddenly pops up here in this PR?

### Edit

It got a bit clearer afterwards, but it prompted me to add a sugg for the option’s description. The relationship between those options should probably be made clear.

### Edit 2

Omg it,s even more complicated than I thought.